tests/unit/stores/pacbioLibraryBatches.spec.js

```diff
@@ -0,0 +1,153 @@
+      expect(pacbioLibraryBatchStore.librariesInBatch).toEqual(
+        pacbioLibraryBatchFactory.storeData.librariesInBatch,
+      )
+    })
+  })
+
+  describe('actions', () => {
+    describe('createLibraryBatch', () => {
+      let create, getTagSets, csvFile, csvFileTextContent, logMessage, mockSuccessResponse, tagSet
+
+      beforeEach(() => {
+        vi.spyOn(apiGetter.traction.pacbio.requests, 'get').mockResolvedValue(
+          pacbioRequestsFactory.responses.fetch,
+        )
+        create = vi.fn()
+        getTagSets = vi.fn().mockResolvedValue(pacbioTagSetFactory.responses.fetch)
+        rootStore.api.v2 = {
+          traction: {
+            pacbio: {
+              library_batches: { create },
+              tag_sets: { get: getTagSets },
+            },
+          },
+        }
+        csvFile = {
+          text: () => Promise.resolve(csvFileTextContent),
+        }
+        logMessage = vi.fn()
+        rootStore.addCSVLogMessage = logMessage
+        mockSuccessResponse = successfulResponse(pacbioLibraryBatchFactory.content)
+        tagSet = pacbioTagSetFactory.storeData.selected.tagSet
+      })
+
+      it('returns error if input params are empty', async () => {
+        const csvFile = {
+          text: () => Promise.resolve(csvFileTextContent),
+        }
+
+        // Test with csvFile as null
+        let result = await pacbioLibraryBatchStore.createLibraryBatch(null, tagSet)
+        expect(result.success).toBeFalsy()
+        expect(result.errors).toEqual(['csvFile is required'])
+
+        // Test with tagSet as null
+        result = await pacbioLibraryBatchStore.createLibraryBatch(csvFile, null)
+        expect(result.success).toBeFalsy()
+        expect(result.errors).toEqual(['tagSet is required'])
+      })
+
+      it('successfully creates a library batch', async () => {
+        csvFileTextContent = fs.readFileSync('./tests/data/csv/pacbio_library_batch.csv', 'utf8')
+        create.mockResolvedValue(mockSuccessResponse)
+        const { success, result } = await pacbioLibraryBatchStore.createLibraryBatch(
+          csvFile,
+          tagSet,
+        )
+        const payload = pacbioLibraryBatchFactory.createLibraryBatchPayloadData(
+          csvFileTextContent,
+          pacbioRequestsFactory.content.data,
+        )
+
+        expect(create).toBeCalled(payload)
+        expect(success).toBeTruthy()
+        expect(pacbioLibraryBatchStore.libraries).toEqual(
+          pacbioLibraryBatchFactory.storeData.libraries,
+        )
+        expect(pacbioLibraryBatchStore.tubes).toEqual(pacbioLibraryBatchFactory.storeData.tubes)
+        expect(result).toEqual(pacbioLibraryBatchStore.librariesInBatch)
+      })
+
+      it('returns error when csv file contains duplicate tags', async () => {
+        csvFileTextContent = fs.readFileSync(
+          './tests/data/csv/pacbio_library_batch_duplicate_tags.csv',
+          'utf8',
+        )
+        create.mockResolvedValue(mockSuccessResponse)
```

Comment:
Don't think this mockResolvedValue is needed

Reply:
I think it is needed as we are checking the result against success.